docs/versioning_vendor_api.md

```diff
@@ -0,0 +1,138 @@
+
+e.g. to make retrieving all applications available in version 1.1
+
+
+in `/app/lib/vendor_api/changes/retrieve_applications.rb`
+
+```ruby
+class RetrieveApplications < VersionChange
+   action ApplicationsController, :index
+end
+```
+
+in `/app/lib/vendor_api,rb`
+
+```ruby
+module VendorAPI
+  ...
+  VERSIONS = {
+   '1.0' => [ ... ],
+   '1.1' => [ Changes::RetrieveApplications ]
+  }
+end
+```
+
+### Adding a new resource
+
+To ensure that the API supports all existing minor versions, any changes must be introduced through presenter modules. This enables serving older versions without including changes introduced at later points.
+
+#### Adding a new presenter
+
+Presenters must extend the `VendorAPI::Base` and return the object Hash through a private `schema` method. The hash can then be converted to JSON through public methods that utilise the application cache (look at `ApplicationPresenter`).
+
+Below, you can see an example of configuring a new `InterviewPresenter`
+
+_configure the new presenter in the VersionChange class_
+
+```ruby
+class AddInterviews < VersionChange
+   resource InterviewPresenter
+end
+```
+
+```ruby
+module VendorAPI
+  ...
+  VERSION = {
+   '1.0' => [...],
+   '1.1' => [AddInterviews],
+  }
+end
+```
+
+#### Updating an existing presenter
+
+When introducing new functionality or changing the behavior of existing fields, create a new module that implements the `schema` method, and deep merge into the returned value. Changing the values directly in the existing presenter is not recommended as that will have an impact on previous versions of the API.
+
+For example, to implement a change in the ApplicationChoice behavior so that a candidate's last name is always returned upcased
+
+
+_introduce a new presenter module..._
+
+```ruby
+module UpdateApplicationCandidateLastName
+  def schema
+    super.deep_merge!({
+      last_name: application.application_form.last_name.upcase
+    })
+  end
+end
+```
+
+_configure this in a version change class_
+
+```ruby
+class UpdateCandidateLastName < VersionChange
+   resource ApplicationsPresenter, [UpdateApplicationCandidateLastName]
+end
+```
+
+_include the change in the vendor API_
+
+```ruby
+module VendorAPI
+  ...
+  VERSIONS = {
+   '1.0' => [...],
+   '1.1' => [...],
+   '1.2' => [UpdateApplicationCandidateLastName]
+  }
+end
+```
+
+
+### Resources
+
+- [Vendor API version information](../app/lib/vendor_api.rb)
+- [Vendor API version change classses](../app/lib/vendor_api/changes)
+- [API Presenter base class](../app/presenters/vendor_api/base.rb)
+
+## Documenting changes
+
+Changes to the API are documented in the release_notes file, after specifying the version and date that the version was released.
+
+For a minor version release
+
+```markdown
+## v1.1 - 19th January 2022
```

Comment:
Bit tempting to autogenerate this — or at least mechanically test it — now that we have changes enumerated!?

Reply:
good question, so we touched on auto generation, as there is ` description` in the change modules, however I think that can be implemented as a separate ticket later and we agreed to keep this light touch for now

Reply:
great